// @autonomys/auto-utils/dist/constants/network.js
"use strict";
// file: src/constants/network.ts
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultNetwork = exports.networks = exports.SUBSCAN_EXPLORER = exports.ASTRAL_EXPLORER = exports.NetworkExplorerName = exports.NetworkName = exports.NetworkId = void 0;
const domain_1 = require("./domain");
const token_1 = require("./token");
var NetworkId;
(function (NetworkId) {
    NetworkId["MAINNET"] = "mainnet";
    NetworkId["TAURUS"] = "taurus";
    NetworkId["GEMINI_3H"] = "gemini-3h";
    NetworkId["DEVNET"] = "devnet";
    NetworkId["LOCALHOST"] = "localhost";
})(NetworkId || (exports.NetworkId = NetworkId = {}));
var NetworkName;
(function (NetworkName) {
    NetworkName["MAINNET"] = "Mainnet";
    NetworkName["TAURUS"] = "Testnet - Taurus";
    NetworkName["GEMINI_3H"] = "Testnet - Gemini 3H";
    NetworkName["DEVNET"] = "Devnet";
    NetworkName["LOCALHOST"] = "Localhost";
})(NetworkName || (exports.NetworkName = NetworkName = {}));
var NetworkExplorerName;
(function (NetworkExplorerName) {
    NetworkExplorerName["ASTRAL"] = "Astral";
    NetworkExplorerName["SUBSCAN"] = "Subscan";
})(NetworkExplorerName || (exports.NetworkExplorerName = NetworkExplorerName = {}));
exports.ASTRAL_EXPLORER = 'https://explorer.autonomys.xyz/';
exports.SUBSCAN_EXPLORER = 'https://subspace.subscan.io/';
exports.networks = [
    {
        id: NetworkId.MAINNET,
        name: NetworkName.MAINNET,
        rpcUrls: ['wss://rpc.mainnet.subspace.foundation/ws'],
        explorer: [
            {
                name: NetworkExplorerName.ASTRAL,
                url: exports.ASTRAL_EXPLORER,
            },
            {
                name: NetworkExplorerName.SUBSCAN,
                url: exports.SUBSCAN_EXPLORER,
            },
        ],
        domains: [],
        token: token_1.DEFAULT_TOKEN,
    },
    {
        id: NetworkId.TAURUS,
        name: NetworkName.TAURUS,
        rpcUrls: [
            'wss://rpc.taurus.subspace.network/ws',
            'wss://rpc-0.taurus.subspace.network/ws',
            'wss://rpc-1.taurus.subspace.network/ws',
        ],
        explorer: [
            {
                name: NetworkExplorerName.ASTRAL,
                url: exports.ASTRAL_EXPLORER + 'taurus/consensus/',
            },
        ],
        domains: [],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
    },
    {
        id: NetworkId.GEMINI_3H,
        name: NetworkName.GEMINI_3H,
        rpcUrls: [
            'wss://rpc-0.gemini-3h.subspace.network/ws',
            'wss://rpc-1.gemini-3h.subspace.network/ws',
        ],
        explorer: [
            {
                name: NetworkExplorerName.ASTRAL,
                url: exports.ASTRAL_EXPLORER + 'gemini-3h/consensus/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['wss://nova-0.gemini-3h.subspace.network/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['wss://autoid-0.gemini-3h.subspace.network/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
    },
    {
        id: NetworkId.DEVNET,
        name: NetworkName.DEVNET,
        rpcUrls: ['ws://rpc.devnet.subspace.network/ws'],
        explorer: [
            {
                name: NetworkExplorerName.ASTRAL,
                url: exports.ASTRAL_EXPLORER + '/devnet/consensus/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['wss:///nova.devnet.subspace.network/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['wss://autoid.devnet.subspace.network/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
        isLocalhost: false,
    },
    {
        id: NetworkId.LOCALHOST,
        name: NetworkName.LOCALHOST,
        rpcUrls: ['ws://127.0.0.1:9944/ws'],
        explorer: [
            {
                name: NetworkExplorerName.ASTRAL,
                url: exports.ASTRAL_EXPLORER + 'localhost/consensus/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['ws:///127.0.0.1:9946/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['ws://127.0.0.1:9945/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
        isLocalhost: true,
    },
];
exports.defaultNetwork = exports.networks[0];
