// @autonomys/auto-utils/dist/constants/network.js
"use strict";
// file: src/constants/network.ts
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultNetwork = exports.networks = exports.ASTRAL_EXPLORER = exports.NetworkId = void 0;
const domain_1 = require("./domain");
const token_1 = require("./token");
var NetworkId;
(function (NetworkId) {
    NetworkId["GEMINI_3H"] = "gemini-3h";
    NetworkId["DEVNET"] = "devnet";
    NetworkId["LOCALHOST"] = "localhost";
})(NetworkId || (exports.NetworkId = NetworkId = {}));
exports.ASTRAL_EXPLORER = 'https://explorer.autonomys.xyz/';
exports.networks = [
    {
        id: NetworkId.GEMINI_3H,
        name: 'Testnet - Gemini 3H',
        rpcUrls: [
            'wss://rpc-0.gemini-3h.subspace.network/ws',
            'wss://rpc-1.gemini-3h.subspace.network/ws',
        ],
        explorer: [
            {
                name: 'Astral',
                url: exports.ASTRAL_EXPLORER + 'gemini-3h/consensus/',
            },
            {
                name: 'Subscan',
                url: 'https://subspace.subscan.io/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['wss://nova-0.gemini-3h.subspace.network/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['wss://autoid-0.gemini-3h.subspace.network/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
    },
    {
        id: NetworkId.DEVNET,
        name: 'Devnet',
        rpcUrls: ['ws://rpc.devnet.subspace.network/ws'],
        explorer: [
            {
                name: 'Astral',
                url: exports.ASTRAL_EXPLORER + '/devnet/consensus/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['wss:///nova.devnet.subspace.network/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['wss://autoid.devnet.subspace.network/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
        isLocalhost: false,
    },
    {
        id: NetworkId.LOCALHOST,
        name: 'Localhost',
        rpcUrls: ['ws://127.0.0.1:9944/ws'],
        explorer: [
            {
                name: 'Astral',
                url: exports.ASTRAL_EXPLORER + 'localhost/consensus/',
            },
        ],
        domains: [
            Object.assign(Object.assign({ domainId: '0' }, domain_1.domains[domain_1.DomainRuntime.NOVA]), { rpcUrls: ['ws:///127.0.0.1:9946/ws'] }),
            Object.assign(Object.assign({ domainId: '1' }, domain_1.domains[domain_1.DomainRuntime.AUTO_ID]), { rpcUrls: ['ws://127.0.0.1:9945/ws'] }),
        ],
        token: token_1.TESTNET_TOKEN,
        isTestnet: true,
        isLocalhost: true,
    },
];
exports.defaultNetwork = exports.networks[0];
